### ISSUE TITLE
Training get NAN loss value when set GPUS>2 using the train_net.py

### ISSUE THREAD

**Issue by JackHenry1992** [roytseng-tw/Detectron.pytorch#19]
Thanks for sharing your greate job.
When I set GPUS=3, batchsize=15, I got `NAN` loss value. Also test GPUS=4 & batchsize=16/20, and the same `NAN` loss. (Training e2e-resnet50-c4-2x from scratch)
![image](https://user-images.githubusercontent.com/9390228/39091945-858391ee-4632-11e8-858a-17f26d096768.png)

![image](https://user-images.githubusercontent.com/9390228/39091574-c86f243c-4628-11e8-8e3a-d8b23fe2a08a.png)

But when using GPUS=2 & batchsize=10 , it will be right
![image](https://user-images.githubusercontent.com/9390228/39091541-4ef5711a-4628-11e8-8663-209ae3e8d8d6.png)

**Comment by roytseng-tw** [comment]
How did set GPUS=3?  Did your change the `NUM_GPUS` option in the config file?
If yes, then actually you **should not** change that value. You only need to set num of gpus through the environment variable `CUDA_VISIBLE_DEVICES`.

The code will linearly scale the learning rate based on the batch size change. Original batch size is infered by `cfg.NUM_GPUS * cfg.TRAIN.IMS_PER_BATCH`, so if you change `cfg.NUM_GPUS` then the original batch size is alter. Hence, learning rate adjustment will go wrong.

**Comment by JackHenry1992** [comment]
The params used for 4 GPUs with 20 batchsize are as follows:

![image](https://user-images.githubusercontent.com/9390228/39092044-cac35e62-4635-11e8-8446-017b95dbf50c.png)

![image](https://user-images.githubusercontent.com/9390228/39092040-bb28007a-4635-11e8-90f7-60038a65c67f.png)

![image](https://user-images.githubusercontent.com/9390228/39092035-a22d7a8c-4635-11e8-8c8e-852e7630e8dc.png)

**Comment by JackHenry1992** [comment]
Solved this by reducing learning rate......
